lib/Dialect/Indexing/IR/Indexing.cpp

```diff
@@ -82,6 +78,12 @@ LogicalResult GatherOp::inferReturnTypes(
   return success();
 }
 
+bool GatherOp::isCompatibleReturnTypes(TypeRange l, TypeRange r) {
```

Comment:
nit: areCompatibleReturnTypes

Reply:
this is an interface implementation https://github.com/llvm/llvm-project/blob/09515f2c20111628ce81ad5f40e12e5f6af5ed2f/mlir/include/mlir/Interfaces/InferTypeOpInterface.td#L89 (so the name isn't up to me, though I could submit an upstream diff...)